docs/components/table-view.md

```diff
@@ -0,0 +1,1011 @@
+          return 1
+        }
+
+        return 0
+      }
+    }
+
+    return 0
+  })
+
+  tableData.value = data
+}
+</script>
+```
+
+:::tip NOTE
+Notice that in the example above the _Username_ column is `sortable` and the _Email_ column is `hidable`.
+:::
+
+#### Reserved Data Keys
+
+:::warning NOTE
+This feature is only available in KTableView and not available in [KTable](/components/table).
+:::
+
+- `to` - if you wish to make the whole row a link, you can pass `to` parameter in each `data` entry. If an object is passed, it will be rendered as `<router-link>` component or otherwise if a string is passed, it will be rendered as an anchor tag. Clicking anywhere on table row will trigger route change. Refer to the example below for usage.
+- `target` - target attribute for row links.
+
+<KTableView
+  :data="tableDataWithLinks"
+  :headers="basicHeaders()"
+/>
+
+```vue
+<template>
+  <KTableView
+    :data="tableData"
+    :headers="headers"
+  />
+</template>
+
+<script setup lang="ts">
+import type { TableViewHeader, TableData } from '@kong/kongponents'
+
+const headers: Array<TableViewHeader> = [
+  {
+    key: 'name', 
+    label: 'Full Name', 
+  },
+  ...
+]
+
+const tableData = ref<TableData>([
+  {
+    id: 1,
+    // notice that property keys in data object correspond to each respective key in headers const
+    name: 'Leanne Graham', 
+    username: 'Bret',
+    email: 'Sincere@april.biz',
+    to: 'https://kongponents.konghq.com/',
+    target: '_blank',
+  },
+  {
+    id: 2,
+    name: 'Ervin Howell',
+    username: 'Antonette',
+    email: 'Shanna@melissa.tv',
+    to: 'https://kongponents.konghq.com/',
+    target: '_blank',
+  },
+  ...
+])
+</script>
+```
+
+### loading
+
+Boolean to control whether the component should display the loading state.
+
+<KTableView
+  loading
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  loading
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### error
+
+Boolean to control whether the component should display the error state. Defaults to `false`. See [error state](#error-1) section for more customization options.
+
+<KTableView
+  error
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  error
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### resizeColumns
+
+Allow table column width to be resizable (defaults to `false`). Adjusting a column's width will trigger an [`update:table-preferences` event](#updatetable-preferences).
+
+<KTableView
+  resize-columns
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```html
+<KTableView
+  resize-columns
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### rowHover
+
+Boolean to control whether table should display hover state upon hovering rows. Defaults to `true`.
+
+### tablePreferences
+
+Table preferences object for persisting some configs.
+
+```ts
+interface TablePreferences {
+  pageSize?: number // the number of items to display per page
```

Comment:
If pagination isn’t included, what value does the page size serve here?

Reply:
This interface is shared between KTable and KTableView and [useTablePreferences composable](https://github.com/Kong/public-ui-components/tree/main/packages/core/core/src/useTablePreferences) uses it.

Reply:
So then can’t we utilize a separate interface that extends the base one…?

Reply:
I think that would be confusing? I'm fine with removing references to these parameters from this docs section but I don't think we should create a separate interface because then we would have to create a new or alter the existing `useTablePreferences` composable. Since those fields are optional anyways, I think we can still use the original interface for both components.

Reply:
You can use the same composable since the property is optional, but for the KTableVoew you can `Omit` the extra property from a similarly-named interface since it doesn't do anything. 

I understand wanting to keep a close parity between components, but inevitably over time they will both need to differ in places since they are fundamentally different under the hood.

Reply:
Now that pagination is included, I suppose we don't need a new interface?

Reply:
Does it utilize _all_ of the same properties? (are there any the new component does not utilize?)

Reply:
Nope, all of them